### ISSUE TITLE
False-positive unused `self` warning

### ISSUE THREAD

**Issue by georgiy-belyanin** [CppCXY/emmylua-analyzer-rust#18]
Try out this code while using a language server.

```lua
local function test(self)
    for k, v in pairs(self) do
        print(k, v)
    end
end
```

LSP says `self` is an unused variable. If you change `self` to a different name, everything works properly, no diagnostics are thrown.

```lua
local function test(sel)
    for k, v in pairs(sel) do
        print(k, v)
    end
end
```

<img width="351" alt="Image" src="https://github.com/user-attachments/assets/12616028-532a-4ecf-a884-0b61826fe3fb" />
<img width="301" alt="Image" src="https://github.com/user-attachments/assets/7c0046bb-3567-43aa-b337-bfe3f3b90158" />

It seems that the reason is behind the special way the `self` identifier is handled. I suppose it should be fixed because sometimes it may be handy to pass something as a `self` argument.
